src/pretix/plugins/sendmail/views.py

```diff
@@ -189,88 +402,111 @@ def form_valid(self, form):
 
-                    self.output[l] = {
-                        'subject': _('Subject: {subject}').format(subject=preview_subject),
-                        'html': preview_text,
-                        'attachment': form.cleaned_data.get('attachment')
-                    }
-
-            self.order_count = ocnt
-            return self.get(self.request, *self.args, **self.kwargs)
+    def describe_match_size(self, cnt):
+        return ngettext(
+            '%(number)s matching order',
+            '%(number)s matching orders',
+            cnt or 0,
+        ) % {
+            'number': intcomma(cnt or 0),
+        }
 
-        kwargs = {
+    def get_task_kwargs(self, form, objects):
+        kwargs = super().get_task_kwargs(form, objects)
+        kwargs.update({
             'recipients': form.cleaned_data['recipients'],
-            'event': self.request.event.pk,
-            'user': self.request.user.pk,
-            'subject': form.cleaned_data['subject'].data,
-            'message': form.cleaned_data['message'].data,
-            'orders': [o.pk for o in orders],
             'items': [i.pk for i in form.cleaned_data.get('items')],
             'not_checked_in': form.cleaned_data.get('not_checked_in'),
             'checkin_lists': [i.pk for i in form.cleaned_data.get('checkin_lists')],
             'filter_checkins': form.cleaned_data.get('filter_checkins'),
             'attach_tickets': form.cleaned_data.get('attach_tickets'),
-        }
-        attachment = form.cleaned_data.get('attachment')
-        if attachment is not None and attachment is not False:
-            kwargs['attachments'] = [form.cleaned_data['attachment'].id]
+        })
+        return kwargs
 
-        send_mails.apply_async(
-            kwargs=kwargs
-        )
-        self.request.event.log_action('pretix.plugins.sendmail.sent',
-                                      user=self.request.user,
-                                      data=dict(form.cleaned_data))
-        messages.success(self.request, _('Your message has been queued and will be sent to the contact addresses of %d '
-                                         'orders in the next few minutes.') % len(orders))
 
-        return redirect(
-            'plugins:sendmail:send',
-            event=self.request.event.slug,
-            organizer=self.request.event.organizer.slug
+class WaitinglistSendView(BaseSenderView):
+    form_class = forms.WaitinglistMailForm
+    form_fragment_name = "pretixplugins/sendmail/send_form_fragment_waitinglist.html"
+    context_parameters = ['event', 'waiting_list_entry', 'event_or_subevent']
+    task = send_mails_to_waitinglist
+
+    ACTION_TYPE = 'pretix.plugins.sendmail.sent.waitinglist'
+    TITLE = _("Waiting list")
+    DESCRIPTION = _("Send an email to every customer who is currently waiting to receive a ticket through the waiting "
+                    "list feature.")
```

Comment:
„Customer“ and „ticket“ are words I would be careful in the context of waitinglists. What about something like:

```suggestion
    DESCRIPTION = _("Send an email to everyone who is currently on the waiting list (ie. has not yet received a voucher).")
```

Reply:
I rephrased it